src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs

```diff
@@ -22,13 +23,22 @@ public class InMemoryExporter<T> : BaseExporter<T>
         where T : class
     {
         private readonly ICollection<T> exportedItems;
+        private readonly Func<Batch<T>, ExportResult> onExport;
 
         public InMemoryExporter(ICollection<T> exportedItems)
         {
             this.exportedItems = exportedItems;
+            this.onExport = (Batch<T> batch) => this.DefaultExport(batch);
         }
 
-        public override ExportResult Export(in Batch<T> batch)
+        public InMemoryExporter(Func<Batch<T>, ExportResult> exportFunc)
```

Comment:
we can leave this internal as we don't expect people to use this directly. (they use it via the extension methods only.)